instrumentation/bunny/lib/opentelemetry/instrumentation/bunny/patch_helpers.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Bunny
+      # The PatchHelper module provides functionality shared between patches.
+      #
+      # For additional details around trace messaging semantics
+      # See https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md#messaging-attributes
+      module PatchHelpers
+        def self.with_send_span(channel, tracer, exchange, routing_key, &block)
+          attributes = basic_attributes(channel, channel.connection, exchange, routing_key)
+          destination = destination_name(exchange, routing_key)
+
+          tracer.in_span("#{destination} send", attributes: attributes, kind: :producer, &block)
+        end
+
+        def self.with_process_span(channel, tracer, delivery_info, properties, &block)
+          destination = destination_name(delivery_info[:exchange], delivery_info[:routing_key])
+          parent_context, links = extract_context(properties)
+
+          OpenTelemetry::Context.with_current(parent_context) do
+            tracer.in_span("#{destination} process", links: links, kind: :consumer, &block)
+          end
+        end
+
+        def self.destination_name(exchange, routing_key)
+          [exchange, routing_key].compact.join('.')
+        end
+
+        def self.extract_context(properties)
+          # use the receive span as parent context
+          parent_context = OpenTelemetry.propagation.extract(properties[:tracer_receive_headers])
+
+          # link to the producer context
+          producer_context = OpenTelemetry.propagation.extract(properties[:headers])
```

Comment:
@fbogsany shouldn't this be `.extract(properties[:headers] || {})`?

If I have two services A & B. The first service is instrumented but the second one is not, then I get this exception in service A:
```
Uncaught exception from consumer #<Bunny::Consumer:70980 @channel_id=2 @queue=amq.gen-p2ZzQp6h4r6QWdCBIC4Pgg> @consumer_tag=bunny-1629126993000-738472400850>: #<NoMethodError: undefined method `[]' for nil:NilClass> @ /home/indrek/.rvm/gems/ruby-2.7.2@admin/gems/opentelemetry-api-1.0.0.rc2/lib/opentelemetry/context/propagation/text_map_getter.rb:16:in `get'
```

I think that's because `properties[:headers]` is `nil` when receiving a message from a service that is not using OTEL.

Reply:
Yes, that seems possible. Could you create a separate issue for this?